feature/platform/transceiver/tests/zr_laser_bias_current_test/zr_laser_bias_current_test.go

```diff
@@ -0,0 +1,218 @@
+	laserbiasInstant := verifyLaserBiasCurrent(t, dut1, pStreamInstant, "laserbiasInstant")
+	t.Logf("laserBias Instant value: %f", laserbiasInstant)
+	laserbiasMin := verifyLaserBiasCurrent(t, dut1, pStreamMin, "laserbiasMin")
+	t.Logf("laserBias Min value: %f", laserbiasMin)
+	laserbiasMax := verifyLaserBiasCurrent(t, dut1, pStreamMax, "laserbiasMax")
+	t.Logf("laserBias Max value: %f", laserbiasMax)
+	laserbiasAvg := verifyLaserBiasCurrent(t, dut1, pStreamAvg, "laserbiasAvg")
+	t.Logf("laserBias Avg value: %f", laserbiasAvg)
+	if laserbiasAvg >= laserbiasMin && laserbiasAvg <= laserbiasMax {
+		t.Logf("The average is between the maximum and minimum values")
+	} else {
+		t.Fatalf("The average is not between the maximum and minimum values Avg:%f Min:%f Max:%f", laserbiasAvg, laserbiasMin, laserbiasMax)
+	}
+}
+func interfaceConfig(t *testing.T, dut1 *ondatra.DUTDevice, dp *ondatra.Port) {
+	d := &oc.Root{}
+	i := d.GetOrCreateInterface(dp.Name())
+	i.Enabled = ygot.Bool(true)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	gnmi.Replace(t, dut1, gnmi.OC().Interface(dp.Name()).Config(), i)
+	OCcomponent := opticalChannelComponentFromPort(t, dut1, dp)
+	gnmi.Replace(t, dut1, gnmi.OC().Component(OCcomponent).OpticalChannel().Config(), &oc.Component_OpticalChannel{
+		TargetOutputPower: ygot.Float64(targetOutputPower),
+		Frequency:         ygot.Uint64(frequency),
+	})
+}
+func TestZRLaserBiasCurrentState(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut")
+	dp1 := dut1.Port(t, "port1")
+	dp2 := dut1.Port(t, "port2")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	interfaceConfig(t, dut1, dp1)
+	interfaceConfig(t, dut1, dp2)
+	intUpdateTime := 2 * time.Minute
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	transceiverState := gnmi.Get(t, dut1, gnmi.OC().Interface(dp1.Name()).Transceiver().State())
+	if dp1.PMD() != ondatra.PMD400GBASEZR {
+		t.Fatalf("%s Transceiver is not 400ZR its of type: %v", transceiverState, dp1.PMD())
+	}
+	OCcomponent := opticalChannelComponentFromPort(t, dut1, dp1)
+	component1 := gnmi.OC().Component(OCcomponent)
+	p1StreamInstant := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Instant().State(), 10*time.Second)
+	p1StreamMin := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Min().State(), 10*time.Second)
+	p1StreamMax := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Max().State(), 10*time.Second)
+	p1StreamAvg := samplestream.New(t, dut1, component1.OpticalChannel().LaserBiasCurrent().Avg().State(), 10*time.Second)
+	verifyLaserBiasCurrentAll(t, dut1, p1StreamInstant, p1StreamAvg, p1StreamMax, p1StreamMin)
+	defer p1StreamAvg.Close()
```

Comment:
I think you'll want to move the defer statements directly underneath where each stream is initialized. Same for throughout the file.

Reply:
updated